akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -137,6 +138,32 @@ object GraphStageLogic {
   private object DoNothing extends (() ⇒ Unit) {
     def apply(): Unit = ()
   }
+
+  /**
+   * Minimal actor to work with other actors and watch them in a synchronous ways
+   */
+  class StageActorRef(asyncCallback: AsyncCallback[(ActorRef, Any)], override val path: ActorPath) extends akka.actor.MinimalActorRef {
+    override def provider: ActorRefProvider =
+      throw new UnsupportedOperationException("GraphStage MinimalActor does not provide")
+
+    override def !(message: Any)(implicit sender: ActorRef = Actor.noSender): Unit = {
+      asyncCallback.invoke((sender, message))
+    }
+
+    // FIXME: This ActorRef is not watchable!
```

Comment:
unfortunately this needs to be fixed. Tcp actors watch these ActorRefs and clean up in the case these go away. I.e. this cannot be postponed to another ticket.